pkg/apis/extensions/types.go

```diff
@@ -332,6 +332,9 @@ type RollingUpdateDeployment struct {
 }
 
 type DeploymentStatus struct {
+	// The generation observed by the deployment controller.
+	ObservedGeneration int `json:"observedGeneration,omitempty"`
```

Comment:
FWIW, `HorizontalPodAutoscalerStatus.ObservedGeneration` uses `*int64`; I don't have strong opinion about whether we use `int64` or `*int64` though.